### PR TITLE
SOC-883 Share forum post to social networks

### PR DESCRIPTION
**Pull request by mklucsarits**
https://wikia-inc.atlassian.net/browse/SOC-883

@lizlux @garthwebb

### CODE REVIEW

**Comment by rogatty** [comment]
There is already a ticket for doing that, see [HG-795](https://wikia-inc.atlassian.net/browse/HG-795).

@inez added a comment against using `window.location` in `ShareFeatureComponent`. Please sync up on this.

File: front/scripts/main/components/ShareFeatureComponent.ts [hunk 1]
```diff
@@ -25,6 +25,8 @@ App.ShareFeatureComponent = Em.Component.extend(App.TrackClickMixin, App.Languag
 		return 0;
 	}),
 
+	sharedUrl: Mercury.wiki.basePath + window.location.pathname,
```
Comment (rogatty) [inline front/scripts/main/components/ShareFeatureComponent.ts]:
```
  TypeError: 'undefined' is not an object (evaluating 'Mercury.wiki.basePath')
  at /var/lib/jenkins/workspace/Mercury-Front/www/front/scripts/main.js:9
```

File: front/scripts/main/routes/DiscussionForumPostRoute.ts [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
 
 App.DiscussionForumPostRoute = Em.Route.extend({
-	model (params: any) {
+	model (params: any): Em.RSVP.Promise {
 		return App.PostModel.find(Mercury.wiki.id, params.postId);
+	},
+	afterModel (model: typeof App.PostModel): void {
+		var title: string = model.get('title');
+		if (!title) {
+			title = 'Discussion on ' + Mercury.wiki.siteName;
```
Comment (inez) [inline front/scripts/main/routes/DiscussionForumPostRoute.ts]: i18n

**Comment by lizlux** [comment]
This LGTM. 

@mklucsarits, did you sort out any outstanding issues with @rogatty and @inez?

**Comment by garthwebb** [comment]
LGTM

File: front/scripts/main/components/ShareFeatureComponent.ts [hunk 3]
```diff
@@ -58,18 +59,18 @@ App.ShareFeatureComponent = Em.Component.extend(App.TrackClickMixin, App.Languag
 
 
 	lineShare: Em.computed('title', function (): string {
-		return "http://line.me/R/msg/text/?" + encodeURIComponent(this.get('title')) + " " + encodeURIComponent(Mercury.wiki.basePath + Mercury.wiki.articlePath + this.get('title'));
+		return 'http://line.me/R/msg/text/?' + encodeURIComponent(this.get('title') + ' ' + this.get('sharedUrl'));
```
Comment (inez) [inline front/scripts/main/components/ShareFeatureComponent.ts]: Are you sure that space ' ' should go through encodeURIComponent? It wasn't going through it before.

File: front/scripts/main/routes/DiscussionPostRoute.ts [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
+	model (params: any): Em.RSVP.Promise {
 		return App.DiscussionPostModel.find(Mercury.wiki.id, params.postId);
 	},
 
+	afterModel (model: typeof App.DiscussionPostModel): void {
+		var title: string = model.get('title');
+		if (!title) {
+			title = i18n.t('discussion.share-default-title', {siteName: Mercury.wiki.siteName});
+		}
+		this.controllerFor('application').set('currentTitle', title);
```
Comment (inez) [inline front/scripts/main/routes/DiscussionPostRoute.ts]: Is this change necessary - seems that it is not being used in sharedUrl propery anyways.

File: front/scripts/main/components/ShareFeatureComponent.ts (hunk 3 above)
Comment (mklucsarits) [inline front/scripts/main/components/ShareFeatureComponent.ts]: Good question. I chose to encode the space because, without it encoded, it's an invalid URL. Even though the web browser would encode it anyway to fix it. The documentation at https://media.line.me/howto/en/ under "Format 1" implies the title and URL that you share, which together make up ```CONTENT_KEY```, should be encoded as a single entity. At least that's how I interpreted it.

File: front/scripts/main/routes/DiscussionPostRoute.ts (hunk 4 above)
Comment (mklucsarits) [inline front/scripts/main/routes/DiscussionPostRoute.ts]: This property is passed into the share-feature component in ```application.hbs```, and is used as the title for sharing on Line.
